### ISSUE TITLE
Fatal error when reading IBuffer return value

### ISSUE THREAD

**Issue by dantler** [NodeRT/NodeRT#44]
I'm attempting to [readValueAsync()](https://msdn.microsoft.com/en-us/library/dn263760.aspx) on a Bluetooth characteristic, but when I check the result, node.exe crashes with this error: `FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.`.

The result is a [GattReadResult](https://msdn.microsoft.com/en-us/library/windows.devices.bluetooth.genericattributeprofile.gattreadresult.aspx).  I can access all the properties normally, except for the most important part: the `value`.  The type is [GattReadResult.Value](https://msdn.microsoft.com/en-us/library/windows.devices.bluetooth.genericattributeprofile.gattreadresult.value.aspx), which is an IBuffer, so I'm trying to create it as a DataReader with the [fromBuffer()](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.streams.datareader.frombuffer) function.  Unfortunately node.exe dies before fromBuffer() can do anything.  Even trying `console.log(result.value);` causes the same crash!

``` javascript

var storage_streams = require('@nodert-win10/windows.storage.streams');
var gattProfile = require('@nodert-win10/windows.devices.bluetooth.genericattributeprofile');

// ... many lines later ...

  characteristic.readValueAsync(function(error, result) {
      if (error) {
          console.error("Failed to read temperature");
          return;
      }
      console.log("Read value result " + result);
      console.log("Read value result proto: " + result.__proto__);
      console.log("Read value result status: " + result.status);
      if (result) {
          console.log("Making data reader...");
          var dataReader = storage_streams.DataReader.fromBuffer(result.value);
          console.log("done"); // This never prints
      }
  });
```

The output looks like this:

```
Read value result [object GattReadResult]
Read value result proto: [object Object]
Read value result status: 0
Making data reader...
FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.
```

And here's a windbg stacktrace:

```
0:000> k 0xc
 # Child-SP          RetAddr           Call Site
00 000000df`51d0e558 00007ffc`be837d9f ntdll!NtTerminateProcess+0x14
01 000000df`51d0e560 00007ffc`bbf9ce6a ntdll!RtlExitUserProcess+0xbf
02 000000df`51d0e590 00007ff7`bd4033a0 KERNEL32!ExitProcessImplementation+0xa
03 000000df`51d0e5c0 00007ff7`bd401e87 node!v8::Unlocker::~Unlocker+0x2e3e30
04 000000df`51d0e630 00007ff7`bcee221f node!v8::Unlocker::~Unlocker+0x2e2917
05 000000df`51d0e690 00007ffc`9b77ac3d node!v8::V8::ToLocalEmpty+0x6f
06 (Inline Function) --------`-------- binding_7ffc9b740000!v8::MaybeLocal<v8::Value>::ToLocalChecked+0xf [c:\...\.node-gyp\4.5.0\include\node\v8.h @ 7164]
07 000000df`51d0e6c0 00007ffc`9b7512ed binding_7ffc9b740000!NodeRT::Utils::CreateExternalWinRTObject+0x23d [c:\...\code\js_ble\node_modules\@nodert-win10\windows.devices.bluetooth.genericattributeprofile\nodertutils.cpp @ 216]
08 000000df`51d0e7a0 00007ffc`9b74186e binding_7ffc9b740000!NodeRT::Windows::Devices::Bluetooth::GenericAttributeProfile::GattReadResult::ValueGetter+0xcd [c:\...\code\js_ble\node_modules\@nodert-win10\windows.devices.bluetooth.genericattributeprofile\_nodert_generated.cpp @ 2665]
09 000000df`51d0e800 00007ff7`bcf106fb binding_7ffc9b740000!Nan::imp::GetterCallbackWrapper+0x9e [c:\...\code\js_ble\node_modules\nan\nan_callbacks_12_inl.h @ 191]
0a 000000df`51d0e840 00007ff7`bcf0ff33 node!v8::Testing::DeoptimizeAll+0xbc7b
0b 000000df`51d0e950 00007ff7`bd001638 node!v8::Testing::DeoptimizeAll+0xb4b3
```

And a failing line of C++ inside NodeRT::Windows::Devices::Bluetooth::GenericAttributeProfile::GattReadResult::ValueGetter (_nodert_generated.cpp @ 2665).
``` c++
info.GetReturnValue().Set(NodeRT::Utils::CreateExternalWinRTObject("Windows.Storage.Streams", "IBuffer", result));
```

**My environment:**

node v4.5.0
node-gyp v3.4.0
Windows 10 14393 x64
Visual Studio 2015 Update 3
Packages:
    "@nodert-win10/windows.devices.bluetooth.genericattributeprofile": "0.2.2",
    "@nodert-win10/windows.devices.enumeration": "0.2.2",
    "@nodert-win10/windows.storage.streams": "0.2.2"

**Comment by nadavbar** [comment]
Hi @dantler, sorry about the delayed answer - 

We've seen a very similar crash in another project, that was caused by the bug in the mechanism that marshals the return value of an object that is not in the functions' namespace.
The problem was fixed it and I've updated the module you've mentioned above. (Bulk updates to all of the modules will come very soon).

The current version is now 0.2.9 - can you please give it a try and see if the issue was resolved?

Also - in case you are working with Streams/Buffers - I'd highly recommend using the <a href="https://github.com/NodeRT/nodert-streams" module, as it let you use WinRT streams just like regular node.js streams.

Let me know if the issue persists, or if there are any other issues.

**Comment by dantler** [comment]
Thanks @nadavbar , updating to 0.2.9 fixed the issue.  I'll close the issue.

I took a quick look at nodert-streams when I was writing my code, but I found the documentation on MSDN to be easier to follow.  To call [writeValueAsync](https://msdn.microsoft.com/en-us/library/dn263773.aspx) I needed to provide an IBuffer and I wasn't sure how to convert a node buffer into an IBuffer for this purpose.  Since there was no example of this in nodert-streams that I could find, I simply clicked around MSDN until I figured it out: I could use a DataWriter to make an IBuffer, so that's what I did.
